README.md

```diff
@@ -17,190 +17,88 @@ By [Fauna, Inc](http://faunadb.com).
-                              Create(ref: Ref("indexes"),
-                                  params: Obj(["name": "posts_by_name",
-                                               "source": BlogPost.classRef,
-                                               "terms": Arr(Obj(["field": Arr("data", "name")])),
-                                               "values": Arr()]))
-                    ))
-        }
-}
+For example, let's create user class...
 
-func rxCreateInstances() -> Observable<Value> {
-    // let's create 100 blog posts using FaunaDB Map expression. Notice `BlogPost` type conforms to `ValueConvertible`.
-    let blogPosts = (1...100).map {
-        BlogPost(name: "Blog Post \($0)", author: "FaunaDB",  content: "content", tags: $0 % 2 == 0 ? ["philosophy", "travel"] : ["travel"])
-    }
-    return client.rx_query(
-                    Map(collection: Arr(blogPosts)) { blogPost  in
-                        Create(ref: Ref("classes/posts"), params: Obj(["data": blogPost]))
-                    })
+```swift
+client.query(Create(ref: Ref("classes"),
+                 params: Obj(["name": "users"])) { createClassR in
+  // handle Fauna DB response
 }
-
-// let's invoke these reactive functions to set up the scheme and create blog post instances.
-rxSetUpSchema(db_name)
-                .flatMap { _ in
-                    rxCreateInstances()
-                }
-                .doOnError { error in
-                  // do something with FaunaDB error }
-                .subscribeNext { value in
-                    // do something with FaunaDB response value
-                }
-                .addDisposableTo(disposeBag)
-
 ```
 
-#### Fauna Client
+And create an user instance...
 
-All communication between your app and FaunaDB db should be done through a Fauna Client, a `Client` instance.
+```swift
+client.query(Create(ref: Ref("classes/users"),
+                 params: Obj(["email": "swift@example.com"]))) { [weak self] result in
+  guard let createR = try? result.dematerialize() else {
+      // something went wrong
+      return
+  }
+  // let's update the recently created user instance
+  self?.client.query(Update(ref: createR.get(field: Field<Ref>("ref")),
+                  params: Obj(["data": Obj(["data": Obj(["name": "Martin",
+                                                         "profession": "dev"])])]))) { resultUpdate in
+  }
+}
+```
 
-The simplest way to set up a client is by passing Fauna DB secret as a parameter.
+now let's see how to paginate over user instances...
 
 ```swift
-let client = Client(secret: <YOUR_FAUNA_DB_SECRET>)
+let match_1 = Match(index: Ref("indexes/users_by_profession"), terms: "dev")
+let match_2 = Match(index: Ref("indexes/users_by_profession"), terms: "dentist")
+let union = Union(sets: match_1, match_2)
+client.query(Paginate(resource: union)) { paginationResult in
+  case .Failure(let error):
+      // handle error
+  case .Success(let value):
```

Comment:
I think it's not the return type that is missing. If we add an example of how to print the user's name on the console, that will be enough to understand how to deal with results.